jdaviz/configs/default/plugins/line_lists/line_list_mixin.py

```diff
@@ -55,3 +58,46 @@ def spectral_lines(self):
     @property
     def available_linelists(self):
         return self.app.get_viewer('spectrum-viewer').available_linelists()
+
+    def set_redshift_slider_bounds(self, range=None, step=None):
+        '''
+        Set the range and/or step of the redshift slider. Set either/both to 'auto'
+        to default based on the limits of the spectrum plot.
+
+        Parameters
+        ----------
+        range : float or `None` or 'auto'
+            Specifies the difference between the upper and lower bounds of the slider.
+            Note that the slider specifies redshift delta from the current value, so a
+            range of 0.1 would allow the user to change the current redshift by +/- 0.05.
+            If `None` or not passed, will leave at the current value. If 'auto',
+            will sync the range based on the limits of the spectrum plot.
+        step : float or `None` or 'auto'
+            Specifies step size of the slider and redshift input (and will be converted to
+            an estimated step for RV). Smaller step sizes will allow finer
+            adjustments/smoother behavior at a potential cost to performance. If `None`
+            or not passed, will leave at the current value. If 'auto', will sync
+            the step size to 1000 steps within the current range.
+        '''
+        if range is not None:
+            msg = RedshiftMessage("rs_slider_range", range, sender=self)
+            self.app.hub.broadcast(msg)
+        if step is not None:
+            msg = RedshiftMessage("rs_slider_step", step, sender=self)
+            self.app.hub.broadcast(msg)
+
+    def set_redshift(self, new_redshift):
+        '''
+        Apply a redshift to any loaded spectral lines and data. Also updates
+        the value shown in the slider.
+        '''
+        if new_redshift == self._redshift:
```

Comment:
Redshift is float, right? Do you want to use `np.allclose` here instead or do you really require exact floating point match?